cl_sii/libs/mm_utils.py

```diff
@@ -99,40 +101,44 @@ def __init__(self, format: Optional[str] = None, **kwargs: Any) -> None:
         self.dateformat = format
 
     def _add_to_schema(self, field_name: str, schema: marshmallow.Schema) -> None:
-        super()._add_to_schema(field_name, schema)
+        super()._bind_to_schema(field_name, schema)
         self.dateformat = self.dateformat or schema.opts.dateformat
 
-    def _serialize(self, value: date, attr: str, obj: object) -> Union[str, None]:
+    def _serialize(
+        self, value: date, attr: str | None, obj: object, **kwargs: Any
+    ) -> Union[str, None]:
         if value is None:
             return None
         self.dateformat = self.dateformat or self.DEFAULT_FORMAT
         format_func = self.DATEFORMAT_SERIALIZATION_FUNCS.get(self.dateformat, None)
         if format_func:
             try:
                 date_str = format_func(value)
-            except (AttributeError, ValueError):
-                self.fail('format', input=value)
+            except (AttributeError, ValueError) as error:
+                raise self.make_error('format', input=value) from error
         else:
             date_str = value.strftime(self.dateformat)
 
         return date_str
 
-    def _deserialize(self, value: str, attr: str, data: dict) -> date:
+    def _deserialize(
+        self, value: str, attr: str | None, data: Mapping[str, Any] | None, **kwargs: Any
+    ) -> date:
         if not value:  # Falsy values, e.g. '', None, [] are not valid
-            self.fail('invalid')
+            raise self.make_error('invalid')
         self.dateformat = self.dateformat or self.DEFAULT_FORMAT
         func = self.DATEFORMAT_DESERIALIZATION_FUNCS.get(self.dateformat)
         if func:
             try:
                 date_value = func(value)  # type: date
-            except (TypeError, AttributeError, ValueError):
-                self.fail('invalid')
+            except (TypeError, AttributeError, ValueError) as error:
+                raise self.make_error('invalid') from error
         elif self.dateformat:
             try:
                 date_value = datetime.strptime(value, self.dateformat).date()
-            except (TypeError, AttributeError, ValueError):
-                self.fail('invalid')
+            except (TypeError, AttributeError, ValueError) as error:
+                raise self.make_error('invalid') from error
```

Comment:
To be consistent with the rest of the library, let's use `as exc` instead of `as error`.